mne/io/brainvision/brainvision.py

```diff
@@ -496,13 +496,35 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
                 if hp_s:
                     info['highpass'] = 1. / info['highpass']
         else:
+            heterogeneous_hp_filter = True
             if hp_s:
+                # We convert channels with disabled filters to having
+                # highpass relaxed / no filters
+                highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
+                            else np.Inf for filt in highpass]
                 info['highpass'] = np.min(np.array(highpass, dtype=np.float))
+                # Coveniently enough 1 / np.Inf = 0.0, so this works for
+                # DC / no highpass filter
                 info['highpass'] = 1. / info['highpass']
+
+                # not exactly the cleanest use of FP, but this makes us
+                # more conservative in *not* warning.
+                if info['highpass'] == 0.0:
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_hp_filter = False
             else:
+                highpass = [float(filt) if filt not in ('NaN', 'Off', 'DC')
+                            else 0.0 for filt in highpass]
                 info['highpass'] = np.max(np.array(highpass, dtype=np.float))
-            warn('Channels contain different highpass filters. Highest filter '
-                 'setting will be stored.')
+                if info['highpass'] == 0.0:
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_hp_filter = False
+            if heterogeneous_hp_filter:
+                warn('Channels contain different highpass filters. '
+                     'Highest filter setting will be stored.')
```

Comment:
> For me too -- in terms of artefacts, better to be conservative by assuming a stronger filter and thus more potential artefacts.

You mean artifacts introduced by filtering, or artifacts suppressed by filtering (e.g., line noise)? The latter seems more important to me, and will also be harmed by the current behavior (i.e., listing `info['lowpass']` as 40 but still getting line artifacts in your data because only some channels were low-pass filtered).